addons/hr_attendance/controllers/main.py

```diff
@@ -3,10 +3,137 @@
+                'attendance': {'check_in': employee.last_attendance_id.check_in,
+                               'check_out': employee.last_attendance_id.check_out},
+                'overtime_today': request.env['hr.attendance.overtime'].sudo().search([
+                    ('employee_id', '=', employee.id), ('date', '=', datetime.date.today()),
+                    ('adjustment', '=', False)]).duration or 0,
+                'use_pin': employee.company_id.attendance_kiosk_use_pin,
+                'display_systray': employee.company_id.attendance_from_systray,
+                'display_overtime': employee.company_id.hr_attendance_display_overtime
+            }
+        return response
+
+    @staticmethod
+    def _get_geoip_response(mode, latitude=False, longitude=False):
+        return {
+            'city': request.geoip.city.name or 'Unknown',
+            'country_name': request.geoip.country.name or request.geoip.continent.name or 'Unknown',
+            'latitude': latitude or request.geoip.location.latitude or False,
+            'longitude': longitude or request.geoip.location.longitude or False,
+            'ip_address': request.geoip.ip,
+            'browser': request.httprequest.user_agent.browser,
+            'mode': mode
+        }
+
+    @http.route('/hr_attendance/kiosk_mode_menu', auth='user', type='http')
+    def kiosk_menu_item_action(self):
+        if request.env.user.user_has_groups("hr_attendance.group_hr_attendance_manager"):
+            # Auto log out will prevent users from forgetting to log out of their session
+            # before leaving the kiosk mode open to the public. This is a prevention security
+            # measure.
+            request.session.logout(keep_db=True)
+            return request.redirect(request.env.user.company_id.attendance_kiosk_url)
+        else:
+            return request.not_found()
+
     @http.route('/hr_attendance/kiosk_keepalive', auth='user', type='json')
     def kiosk_keepalive(self):
         request.session.touch()
         return {}
+
+    @http.route(["/hr_attendance/<token>"], type='http', auth='public', website=True, sitemap=True)
+    def open_kiosk_mode(self, token):
+        company = self._get_company(token)
+        if not company:
+            return request.not_found()
+        else:
+            employee_list = [{"id": e.id,
+                              "name": e.name,
+                              "avatar": e.avatar_1024.decode(),
+                              "job": e.job_id.name,
+                              "department": {"id": e.department_id.id, "name": e.department_id.name}} for e in request.env['hr.employee'].sudo().search([('company_id', '=', company.id)])]
+            departement_list = [{'id': dep.id,
+                                 'name': dep.name,
+                                 'count': dep.total_employee} for dep in request.env['hr.department'].sudo().search([('company_id', '=', company.id)])]
```

Comment:
Nitpicking. I would rather use search_read to avoid prefetching most of hr.employee fields (should be ok for department as well)
```suggestion
            employee_list = [{
                "id": e.id,
                "name": e.name,
                "avatar": e.avatar_1024.decode(),
                "job": e.job_id.name,
                "department": {
                    "id": e.department_id.id,
                    "name": e.department_id.name
                }
            } for e in request.env['hr.employee'].sudo().search([('company_id', '=', company.id)])]
            departement_list = [{
                'id': dep.id,
                'name': dep.name,
                'count': dep.total_employee
            } for dep in request.env['hr.department'].sudo().search([('company_id', '=', company.id)])]
```